### PR TITLE
Add --ignore-running flag with pipelinrun and taskrun delete commands.

### LINKED ISSUE #1432: --ignore-running with pipelinerun/taskrun delete
### Feature request
add **--ignore-running**  flag with pipelinrun and taskrun delete commands.
Currently, we can not ignore the running taskrun/pipelinerun with the delete command.

### Use case
Adding this will help delete all the pipelineruns/taskruns ignoring the running ones.
Combined with other flags would give the user more precise resource deletion

### UI Example
tkn taskrun delete --ignore-running
tkn pipelinerun delete --ignore-running

### PR DESCRIPTION
**Pull request by liangyuanpeng**
# Changes

Fixed #1432 

Add --ignore-running flag with pipelinrun and taskrun delete commands.  

Ignore the running taskrun/pipelinerun when run delete command.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
I wonder if the default value should be `true` 👼🏼.

**Comment by liangyuanpeng** [comment]
> I wonder if the default value should be `true` 👼🏼.

You're right,It should be.

File: pkg/cmd/taskrun/delete_test.go [hunk 1]
```diff
@@ -432,6 +432,15 @@ func TestTaskRunDelete(t *testing.T) {
...
 			want:        "Are you sure you want to delete all TaskRuns related to ClusterTask \"random\" keeping 1 TaskRuns (y/n): All but 1 TaskRuns associated with ClusterTask \"random\" deleted in namespace \"ns\"\n",
 		},
+		{
+			name:        "Delete all with --ignore-running",
+			command:     []string{"delete", "--all", "-f", "-n", "ns", "--ignore-running"},
+			dynamic:     seeds[4].dynamicClient,
+			input:       seeds[4].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "All TaskRuns deleted in namespace \"ns\"\n",
```
Comment (vinamra28) [inline pkg/cmd/taskrun/delete_test.go:442]: should we handle the error message for this ? eg `All non-running TaskRuns have been deleted in namespace ns` :thinking:
Also can you please a test case where one taskrun is in running state and one in pending and after executing the command should delete the one which is not running?

File: pkg/cmd/pipelinerun/delete_test.go [hunk 2]
```diff
@@ -358,6 +358,15 @@ func TestPipelineRunDelete(t *testing.T) {
...
 			want:        "Are you sure you want to delete all PipelineRuns in namespace \"ns\" (y/n): All PipelineRuns deleted in namespace \"ns\"\n",
 		},
+		{
+			name:        "Delete all by ignore-running",
+			command:     []string{"rm", "--all", "-n", "ns", "--ignore-running"},
+			dynamic:     seeds[5].dynamicClient,
+			input:       seeds[5].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "Are you sure you want to delete all PipelineRuns in namespace \"ns\" (y/n): All PipelineRuns deleted in namespace \"ns\"\n",
```
Comment (vinamra28) [inline pkg/cmd/pipelinerun/delete_test.go:368]: same as above

File: pkg/cmd/pipelinerun/delete_test.go [hunk 3]
```diff
@@ -358,6 +358,15 @@ func TestPipelineRunDelete(t *testing.T) {
 			wantError:   false,
 			want:        "Are you sure you want to delete all PipelineRuns in namespace \"ns\" (y/n): All PipelineRuns deleted in namespace \"ns\"\n",
 		},
+		{
+			name:        "Delete all by ignore-running",
```
Comment (piyush-garg) [inline pkg/cmd/pipelinerun/delete_test.go:362]: can you please add more test scenarios, like this behavior should be the same without flag also, and it is not deleting pipeline when setting the flag false? The different status you can try

File: pkg/cmd/taskrun/delete.go [hunk 4]
```diff
@@ -184,11 +185,26 @@ func taskRunLister(p cli.Params, keep int, kind string, cs *cli.Clients) func(st
...
 	taskRuns, err := trlist.TaskRuns(cs, metav1.ListOptions{}, ns)
 	if err != nil {
 		return nil, err
 	}
+
+	if ignoreRunning {
+		var taskRunTmp = []v1beta1.TaskRun{}
+		for _, v := range taskRuns.Items {
+			for _, v2 := range v.Status.Conditions {
+				if v2.Reason == "Running" || v2.Reason == "Pending" {
```
[the remark this answers is not in the document]
Comment (piyush-garg) [reply pkg/cmd/taskrun/delete.go:198]: I think we should consider the `Pending` case also

**Comment by pradeepitm12** [comment]
Hey @liangyuanpeng  :wave: 
Thanks for putting your efforts and raising this pr.
Just wanted to check if you would be able to address the review comments any soon.
We would love to this feature in the coming release.

**Comment by pradeepitm12** [comment]
/close 
with ref: https://github.com/tektoncd/cli/pull/1445
Created a copy of the pr, as required for the upcoming release.

**Pull request closed without merging**